nrf52: Add USB peripheral driver

bergzand opened the pull request:
### Contribution description

This PR adds a USB peripheral driver for the nrf52840 MCU.

Makefile configuration is supplied for the nrf52840dk and the nrf52840-mdk. Feel free to suggest more boards if I missed some.

### Testing procedure

**WARNING**: due to errata 104 and errata 180, this driver won't work for the engineering sample A chips. From engineering sample B and newer it should be fine.

### Issues/PRs references

depends on #9830, requires #10916 for actual testing

![image](https://bergzand.net/misc/usbus.dot.svg)

miri64 commented:
Needs rebase, I believe ;-)

bergzand commented:
> Needs rebase, I believe ;-)

Also needs a bit of work, I've neglected so far to rework this driver to the current periph_usbdev architecture.

dylad commented:
BTW, I have a nrf52dk so I should be able to help here :)

bergzand commented:
rebased and updated!

bergzand commented:
Ready for review here

dylad commented on cpu/nrf52/include/nrfusb.h:
```diff
@@ -0,0 +1,78 @@
+#ifdef __cplusplus
+}
+#endif
+#endif /* SAM_USB_H */
```
Not SAM anymore

bergzand replied:
Oh, :cry: 
fixed

dylad commented on cpu/nrf52/periph/usbdev.c:
```diff
@@ -0,0 +1,654 @@
+ */
+
+/**
+ * @ingroup sam0_common
```
not sam0

bergzand replied:
Fixed

dylad commented on cpu/nrf52/periph/usbdev.c:
```diff
@@ -0,0 +1,654 @@
+ */
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
```
I didn't see any errno, maybe this can be removed ?

bergzand replied:
Required for the `-ENOTSUP` return values

bergzand commented:
>  Is there any restriction with nRF52832 devices ?? I see a bunch of reference to nRF52840 but does it work on nRF52832 ??

As far as I know, the nRF52832 doesn't have the USB peripheral, only the nRF52840 has one.

bergzand commented:
@aabadie Would you mind taking a look here, @dylad doesn't have an nrf52840 based board available for testing?

aabadie commented:
> the nRF52832 doesn't have the USB peripheral, only the nRF52840 has one

I checked the datasheet of both and this is indeed the case. nrf51 also doesn't provide an USB peripheral.

aabadie commented:
I can test on nrf52840-dk/mdk.

bergzand commented:
Squashed and rebased.

@aabadie I took the liberty of fixing the alphabetical order of the `Makefile.features` for the nrf52840-mdk board while fixing the merge conflict.

bergzand commented:
And fixed two typo's in two commit messages

aabadie commented:
> I took the liberty of fixing the alphabetical order of the Makefile.features for the nrf52840-mdk board while fixing the merge conflict.

Thanks @bergzand, I overlooked this when reviewing #11513.

Now that everything is green, let's merge this one. Awesome work :)
